Bottom: 18a148a7a30fc0ee0f7d620670000cc484f13b13
Top:    3634db1ea5c58d1498f994500fd8ad6f774899e0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-22 15:05:55 +0100

Refresh of major-re-write-to-drawing-rout

---

diff --git a/src/draw.c b/src/draw.c
index efa905f..f28f6c9 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -88,7 +88,7 @@ static const BoxType *clip_box = NULL;
  */
 static void Redraw (Boolean, BoxTypePtr);
 static void DrawEverything (BoxTypePtr);
-void DrawTop (const BoxType *);
+static void DrawTop (const BoxType *);
 static int DrawLayerGroup (int, const BoxType *);
 /* static */ void DrawPinOrViaLowLevel (PinTypePtr, Boolean);
 static void ClearOnlyPin (PinTypePtr, Boolean);
@@ -108,7 +108,7 @@ static void SetPVColor (PinTypePtr, int);
 /* static */ void DrawEMark (ElementTypePtr, LocationType, LocationType, Boolean);
 static void ClearPad (PadTypePtr, Boolean);
 /* static */ void DrawHole (PinTypePtr);
-/* static */ void DrawMask (BoxType *);
+static void DrawMask (BoxType *);
 /* static */ void DrawRats (BoxType *);
 /* static */ void DrawSilk (int, int, BoxType *);
 static int pin_callback (const BoxType * b, void *cl);
@@ -760,7 +760,7 @@ DrawSilk (int new_swap, int layer, BoxTypePtr drawn_area)
 /* ---------------------------------------------------------------------------
  * draws solder mask layer - this will cover nearly everything
  */
-/* static */ void
+static void
 DrawMask (BoxType * screen)
 {
   struct pin_info info;
diff --git a/src/draw.h b/src/draw.h
index 0acb0cb..d7a2774 100644
--- a/src/draw.h
+++ b/src/draw.h
@@ -82,9 +82,10 @@ void DrawPlainVia (PinTypePtr, Boolean);
 void DrawRegularText (LayerTypePtr, TextTypePtr, int);
 void DrawEMark (ElementTypePtr, LocationType, LocationType, Boolean);
 void DrawHole (PinTypePtr);
-void DrawMask (BoxType *);
 void DrawRats (BoxType *);
 void DrawSilk (int, int, BoxType *);
+void DrawSpecialPolygon (HID * hid, hidGC DrawGC, LocationType X, LocationType Y, int Thickness);
+
 /* TEMPORARY */
 
 /*GdkDrawable *draw_get_current_drawable(void);*/
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 3dfd9ac..efed6ef 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -46,6 +46,7 @@
 #include "error.h"
 #include "misc.h"
 #include "set.h"
+#include "rtree.h"
 #include "snavi.h"
 #include "gui-trackball.h"
 
@@ -965,7 +966,7 @@ struct pin_info
 {
   Boolean arg;
   LayerTypePtr Layer;
-  BoxTypePtr drawn_area;
+  const BoxType *drawn_area;
 };
 
 void
@@ -1094,10 +1095,10 @@ pad_callback (const BoxType * b, void *cl)
 static int
 hole_callback (const BoxType * b, void *cl)
 {
+#if 0
   PinTypePtr pin = (PinTypePtr) b;
   int plated = cl ? *(int *) cl : -1;
 
-#if 0
   switch (plated)
     {
     case -1:
@@ -1146,7 +1147,7 @@ text_callback (const BoxType * b, void *cl)
 }
 
 static void
-DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon, BoxTypePtr drawn_area)
+DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon, const BoxType *drawn_area)
 {
   static char *color;
 
@@ -1472,13 +1473,13 @@ hole_cyl_callback (const BoxType * b, void *cl)
   PinTypePtr Pin = (PinTypePtr) b;
   struct cyl_info *info = cl;
   DrawDrillChannel (Pin->X, Pin->Y, Pin->DrillingHole / 2, info->from_layer, info->to_layer, info->scale);
+  return 0;
 }
 
 void
 ghid_draw_everything (BoxTypePtr drawn_area)
 {
-  int i, ngroups, side;
-  int plated;
+  int i, ngroups;
   /* This is the list of layer groups we will draw.  */
   int do_group[MAX_LAYER];
   /* This is the reverse of the order in which we draw them.  */
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index aa5a6c1..81e0c09 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -561,6 +561,7 @@ GdkPixmap *ghid_render_pixmap (int cx,
 /* gtkhid-main.c */
 
 void ghid_invalidate_all ();
+void ghid_global_alpha_mult (hidGC gc, double alpha_mult);
 void ghid_invalidate_current_gc ();
 void ghid_get_coords (const char *msg, int *x, int *y);
 gint PCBChanged (int argc, char **argv, int x, int y);
